osu.Game/Rulesets/Edit/IPositionSnapProvider.cs

```diff
@@ -1,10 +1,12 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Allocation;
 using osuTK;
 
 namespace osu.Game.Rulesets.Edit
 {
+    [Cached]
```

Comment:
As above, children of a component which inherits `IDistanceSnapProvider` will not be able to DI `IPositionSnapProvider`.
```suggestion
    [Cached(typeof(IPositionSnapProvider))]
```

Should be applied to all interface types for correctness sake.

Reply:
Failed in zero cases though, are you sure about this?

Reply:
Hmm, right. Seems this is actually not the case for interfaces because they don't have the concept of inheritance, but they're grouped all under a list of interfaces when fetching them from the class type (i.e. `typeof(OsuHitObjectComposer).GetInterfaces() -> new[] { IDistanceSnapProvider, IPositionSnapProvider }`).

Fair enough I guess, it does mean that `CachedAttribute` is yet again working in a weird way? dunno.